lib/postgrex/protocol.ex

```diff
@@ -654,18 +665,28 @@ defmodule Postgrex.Protocol do
     end
   end
 
+  defp execute_send( s, status, %Stream{ state: :suspended, portal: portal, max_rows: max_rows, query: query } = stream, _params, buffer ) do
+    %Query{ name: name } = query
+    messages = [ msg_execute(name_port: portal, max_rows: max_rows) | sync_msgs( status, name ) ]
+    send_and_recv( s, status, stream, buffer, messages, &execute_recv/4 )
+  end
+
+  defp execute_send( s, status, %Stream{ portal: portal, max_rows: max_rows, query: query } = stream, params, buffer ) do
+    %Query{param_formats: pfs, result_formats: rfs, name: name} = query
+    messages = [
+      msg_bind(name_port: portal, name_stat: name, param_formats: pfs, params: params, result_formats: rfs),
```

Comment:
`name` may be `""` for un-named or non-empty for a named statement. In either case the query may not have been prepared on the connection. See logic for executing a query.